tests/tests_pytorch/loops/test_loops.py

```diff
@@ -786,29 +806,58 @@ def _get_iterator(self):
     train_dataloader = TestDataLoader(RandomDataset(32, 64), num_workers=1, persistent_workers=persistent_workers)
     val_dataloader = TestDataLoader(RandomDataset(32, 64), num_workers=1, persistent_workers=persistent_workers)
 
-    class TestCallback(Callback):
-        def on_train_epoch_end(self, trainer, *_):
-            if trainer.current_epoch == 1:
-                raise CustomException
-
-    max_epochs = 3
-
-    model = BoringModel()
-    trainer = Trainer(
-        default_root_dir=tmpdir,
-        limit_train_batches=2,
-        limit_val_batches=2,
-        max_epochs=max_epochs,
-        callbacks=TestCallback() if should_fail else None,
-    )
-
     if should_fail:
         with pytest.raises(CustomException):
             trainer.fit(model, train_dataloader, val_dataloader)
     else:
         trainer.fit(model, train_dataloader, val_dataloader)
 
-    assert train_dataloader.count_shutdown_workers == 2 if should_fail else (2 if persistent_workers else max_epochs)
-    # on sanity checking end, the workers are being deleted too.
-    expected = 2 if persistent_workers else (3 if should_fail else max_epochs + 1)
-    assert val_dataloader.count_shutdown_workers == expected
+    if persistent_workers:
+        expected = [trainer.current_epoch, trainer.current_epoch]  # once epoch end, once on teardown
+    elif should_fail:
+        expected = [
+            # iterable check
+            0,
```

Comment:
This extra shutdown call is caused by the added iterable checks. This is the only change from master in this test

This means that checking whether the input dataloader is iterable does have a performance impact.